### DISCUSSION TITLE
Trigger query after pushState?

### DISCUSSION THREAD

**Discussion by FredrikEkstroem** [microsoft-search/pnp-modern-search#1441] [Q&A]
Hi,
Version: 4.4.0

I'm having a query template that simplified is looking something like this
```typescript
{searchTerms} {?Country:{QueryString.Country} }
```
it works perfectly fine when going to the url directly, but it's not triggering when url is updated using 
```typescript
window.history.pushState(history.state, document.title, newUrl);
```
unless {searchTerms} (also connected to query string) is updated. 

Is there a way to trigger a re render of the search results web part?

**Comment by wobba** [comment]
Can you log this as a bug with complete repro steps for web parts on the page and how they are set up?

**Reply by FredrikEkstroem** [reply]
Bug added #1444

Btw, there is no way to intercept the search query before it's sent or extend with custom tokens for use in query template?

**Reply by FranckyC** [reply]
Short answer, no. At least from a custom component and the provided extensibility framework. Did you try this by any chance https://docs.microsoft.com/en-us/sharepoint/dev/spfx/building-search-extensions?
